src/ekat/ekat_parameter_list.hpp

```diff
@@ -113,6 +117,19 @@ inline void ParameterList::set (const std::string& name, const T& value) {
   }
 }
 
+template<typename T>
+inline bool ParameterList::isType (const std::string& name) {
+  // Try to get variable with type T, return true if successful, false if an error is thrown
+  try {
+    auto& dummy = any_cast<T>(m_params.at(name));
```

Comment:
Before entering the try block, we should check that the parameter exists. If it doesn't, we should throw and notify the user.

Note: it must be done outside the try block, cause otherwise the exception we throw would be caught by the catch block.